### ISSUE TITLE
autotools: some standard targets don't build properly

### ISSUE THREAD

**Issue by sim590** [savoirfairelinux/opendht#129]
For e.g., if you do the following:

```
$ ./autogen.sh && ./configure && make dist
$ cd opendht-$version
$ ./configure
```

you end up with:

```
config.status: error: cannot find input file: `doc/Doxyfile.in'
```

Also, `make dist-*` are not honored. For e.g.:

```
make dist-xz
...
make[1]: *** No rule to make target 'distdir'.  Stop.
```

These options are standard on GNU systems and should be implemented.

**Comment by aberaud** [comment]
Fixed by 65bdfaf22bbc526e26aee78d86052535332af16f

**Comment by sim590** [comment]
@aberaud: I still get this error:
```
make[1]: *** No rule to make target 'distdir'.  Stop.
```

**Comment by sim590** [comment]
As I have validated with @aberaud, it seems that we both get diffrent results. I can confirm that the issue happens despite in a clean repository.

I've just figured out from a clean repository that there is no `Makefile` produced from the command `./configure --enable-python`, but I can see `Makefile.am` and `Makefile.in`. So, I have checked the output of `./configure` and the reason why there's no `Makefile` in the repository is that `cython` was not found on my system (because of naming convention `cython` vs `cython3`).

Secondly, whether cython is found/enabled or not shouldn't result in `make dist` failing.
